crates/turbopack-dev-server/src/source/conditional.rs

```diff
@@ -41,25 +43,31 @@ impl ConditionalContentSourceVc {
 impl ContentSource for ConditionalContentSource {
     #[turbo_tasks::function]
     async fn get(
-        &self,
+        self_vc: ConditionalContentSourceVc,
         path: &str,
         data: turbo_tasks::Value<ContentSourceData>,
     ) -> Result<ContentSourceResultVc> {
-        let first = self.activator.get(path, data.clone());
-        if let ContentSourceContent::NotFound = &*first.await?.content.await? {
-            if !*self.activated.get() {
-                return Ok(first);
-            }
+        let this = self_vc.await?;
+        if !*this.activated.get() {
```

Comment:
Or does this refactor solve the performance penalty by not activating the main source yet?